Mathlib/Analysis/InnerProductSpace/JointEigenspace.lean

```diff
@@ -28,85 +34,108 @@ and a proof obligation that the basis vectors are eigenvectors.
+  · exact (Submodule.mem_iInf <| fun _ ↦ eigenspace (T _) (g _)).mp Eg.2 _
 
 variable [FiniteDimensional 𝕜 E]
 
+open IsFinitelySemisimple
+
 /-- If A and B are commuting symmetric operators on a finite dimensional inner product space
-then the eigenspaces of the restriction of B to any eigenspace of A exhaust that eigenspace.-/
-theorem iSup_eigenspace_inf_eigenspace (hB : B.IsSymmetric)
-    (hAB : A ∘ₗ B = B ∘ₗ A):
+then the eigenspaces of the restriction of B to any eigenspace of A exhaust that eigenspace. -/
+theorem iSup_eigenspace_inf_eigenspace_of_commute (hB : B.IsSymmetric) (hAB : Commute A B) :
     (⨆ γ, eigenspace A α ⊓ eigenspace B γ) = eigenspace A α := by
   conv_rhs => rw [← (eigenspace A α).map_subtype_top]
-  simp only [eigenspace_inf_eigenspace hAB, ← Submodule.map_iSup]
+  simp only [← genEigenspace_eq_eigenspace (f := B), ← Submodule.map_iSup,
+    (eigenspace A α).inf_genEigenspace _ (mapsTo_genEigenspace_of_comm hAB α 1)]
   congr 1
-  rw [← Submodule.orthogonal_eq_bot_iff]
-  exact orthogonalComplement_iSup_eigenspaces_eq_bot <|
-    hB.restrict_invariant <| eigenspace_invariant_of_commute hAB α
+  simpa only [genEigenspace_eq_eigenspace, Submodule.orthogonal_eq_bot_iff]
+    using orthogonalComplement_iSup_eigenspaces_eq_bot <|
+      hB.restrict_invariant <| mapsTo_genEigenspace_of_comm hAB α 1
 
 /-- If A and B are commuting symmetric operators acting on a finite dimensional inner product space,
 then the simultaneous eigenspaces of A and B exhaust the space. -/
-theorem iSup_iSup_eigenspace_inf_eigenspace_eq_top (hA : A.IsSymmetric) (hB : B.IsSymmetric)
-    (hAB : A ∘ₗ B = B ∘ₗ A) :
+theorem iSup_iSup_eigenspace_inf_eigenspace_eq_top_of_commute (hA : A.IsSymmetric)
+    (hB : B.IsSymmetric) (hAB : Commute A B) :
     (⨆ α, ⨆ γ, eigenspace A α ⊓ eigenspace B γ) = ⊤ := by
-  simpa [iSup_eigenspace_inf_eigenspace hB hAB] using
+  simpa [iSup_eigenspace_inf_eigenspace_of_commute hB hAB] using
     Submodule.orthogonal_eq_bot_iff.mp <| hA.orthogonalComplement_iSup_eigenspaces_eq_bot
 
 /-- Given a commuting pair of symmetric linear operators on a finite dimensional inner product
 space, the space decomposes as an internal direct sum of simultaneous eigenspaces of these
 operators. -/
-theorem directSum_isInteral_of_commute (hA : A.IsSymmetric) (hB : B.IsSymmetric)
-    (hAB : A ∘ₗ B = B ∘ₗ A) :
+theorem directSum_isInternal_of_commute (hA : A.IsSymmetric) (hB : B.IsSymmetric)
+    (hAB : Commute A B) :
     DirectSum.IsInternal (fun (i : 𝕜 × 𝕜) ↦ (eigenspace A i.2 ⊓ eigenspace B i.1)):= by
   apply (orthogonalFamily_eigenspace_inf_eigenspace hA hB).isInternal_iff.mpr
   rw [Submodule.orthogonal_eq_bot_iff, iSup_prod, iSup_comm]
-  exact iSup_iSup_eigenspace_inf_eigenspace_eq_top hA hB hAB
-
-end Pair
+  exact iSup_iSup_eigenspace_inf_eigenspace_eq_top_of_commute hA hB hAB
+
+/-- In finite dimensions, the indexed supremum of the joint eigenspaces of a commuting family
+of symmetric linear operators equals `⊤`. -/
```

Comment:
Optional: I know you liked this language before, and it *does* match the descriptions we use for triangularizable endomorphisms in the other file:
```suggestion
/-- A commuting family of symmetric linear maps on a finite dimensional inner
product space is simultaneously diagonalizable. -/
```